### PR TITLE
feat(api): add V1 suggestions API endpoints

### PR DESCRIPTION
**Pull request by alecf**
## Summary

- Add V1-style suggestions API with consistent response shapes matching other V1 endpoints
- `GET /v1/threads/:threadId/messages/:messageId/suggestions` - List suggestions for a message
- `POST /v1/threads/:threadId/messages/:messageId/suggestions` - Generate suggestions for a message

### Key changes from beta API

| Aspect | Beta | V1 |
|--------|------|-----|
| Empty suggestions | 404 error | 200 with empty array |
| Field name | `detailedSuggestion` | `description` |
| Response shape | Raw array | `{ suggestions, hasMore, nextCursor }` |
| Timestamps | Not exposed | `createdAt` (ISO 8601) |

### Files changed

- **`apps/api/src/v1/dto/suggestion.dto.ts`** - New V1 suggestion DTOs
- **`apps/api/src/v1/v1.controller.ts`** - Added GET/POST suggestion endpoints
- **`apps/api/src/v1/v1.service.ts`** - Added listSuggestions, generateSuggestions methods
- **`apps/api/src/threads/threads.service.ts`** - Made createTamboBackendForThread public
- **`packages/db/src/operations/suggestions.ts`** - Added listSuggestionsPaginated

## Test plan

- [x] 12 unit tests covering happy paths and edge cases
- [x] Type checking passes
- [x] Lint passes
- [ ] Manual testing of endpoints via Swagger UI

🤖 Generated with [Claude Code](https://claude.ai/claude-code)

### CODE REVIEW

File: apps/api/src/v1/dto/suggestion.dto.ts [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    description: "Available components that suggestions can reference",
+    type: [V1AvailableComponentDto],
+    required: false,
+  })
+  @IsOptional()
+  @IsArray()
+  @ValidateNested({ each: true })
+  @Type(() => V1AvailableComponentDto)
+  availableComponents?: V1AvailableComponentDto[];
+}
```
[the remark this answers is not in the document]
Comment (alecf) [reply apps/api/src/v1/dto/suggestion.dto.ts:106]: fixed with claude

File: apps/api/src/v1/v1.controller.ts [hunk 2]
```diff
@@ -581,4 +586,80 @@ export class V1Controller {
...
+    status: 404,
+    description: "Message not found",
+  })
+  async generateSuggestions(
+    @Param("threadId") threadId: string,
+    @Param("messageId") messageId: string,
+    @Body() dto: V1GenerateSuggestionsDto,
+  ): Promise<V1ListSuggestionsResponseDto> {
+    return await this.v1Service.generateSuggestions(threadId, messageId, dto);
+  }
```
[the remark this answers is not in the document]
Comment (alecf) [reply apps/api/src/v1/v1.controller.ts:696]: fixed with claude

[the remark this answers is not in the document]
Comment (alecf) [reply apps/api/src/v1/v1.controller.ts:696]: fixed with claude

File: apps/api/src/v1/v1.service.ts [hunk 3]
```diff
@@ -1076,4 +1086,191 @@ export class V1Service {
...
+      nextCursor: hasMore
+        ? encodeV1CompoundCursor({
+            createdAt:
+              resultSuggestions[resultSuggestions.length - 1].createdAt,
+            id: resultSuggestions[resultSuggestions.length - 1].id,
+          })
+        : undefined,
+      hasMore,
+    };
+  }
```
[the remark this answers is not in the document]
Comment (alecf) [reply apps/api/src/v1/v1.service.ts:1193]: fixed with claude

[the remark this answers is not in the document]
Comment (alecf) [reply apps/api/src/v1/v1.service.ts:1193]: fixed with claude

File: apps/api/src/v1/v1.service.ts [hunk 4]
```diff
@@ -1076,4 +1086,191 @@ export class V1Service {
...
+        }
+
+        const contextKey = thread.contextKey ?? "anonymous";
+
+        // Get thread messages for context and convert to ThreadMessage format
+        const dbMessages = await operations.getMessages(this.db, threadId);
+        const threadMessages = dbMessages.map((m) =>
+          dbMessageToThreadMessage(m),
+        );
+
```
[the remark this answers is not in the document]
Comment (alecf) [reply apps/api/src/v1/v1.service.ts]: fixed with claude, set limit to 5

File: apps/api/src/v1/v1.service.ts [hunk 5]
```diff
@@ -1076,4 +1086,191 @@ export class V1Service {
...
+        this.logger.log(
+          `Generated ${savedSuggestions.length} suggestions for message ${messageId}`,
+        );
+
+        return {
+          suggestions: savedSuggestions.map((s) => this.suggestionToDto(s)),
+          hasMore: false,
+        };
+      },
+    );
```
[the remark this answers is not in the document]
Comment (alecf) [reply apps/api/src/v1/v1.service.ts:1327]: fixed with claude

File: packages/db/src/operations/suggestions.ts [hunk 6]
```diff
@@ -1,13 +1,63 @@
...
+    if (cursorCondition) {
+      conditions.push(cursorCondition);
+    }
+  }
+
+  return await db.query.suggestions.findMany({
+    where: and(...conditions),
+    orderBy: [asc(suggestions.createdAt), asc(suggestions.id)],
+    limit,
+  });
```
[the remark this answers is not in the document]
Comment (alecf) [reply packages/db/src/operations/suggestions.ts:66]: fixed with claude
